### PR TITLE
Feature/follow type

### PR DESCRIPTION
**Pull request by cescoferraro**
solves #16

### CODE REVIEW

**Comment by cescoferraro** [comment]
@zmb3 can you take a look at this?

File: user_test.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func TestFollowUsersMissingScope(t *testing.T) {
 	client := testClientString(http.StatusForbidden, json)
 	addDummyAuth(client)
 
-	err := client.Follow(ID("exampleuser01"))
+	err := client.FollowUser(ID("exampleuser01"))
```
Comment (zmb3) [inline user_test.go]: In the tests we should verify that the `type` is set correctly.

There should be a `getLastRequest ()` that will give you the HTTP request that was made, and then you can inspect its URL values at `req.URL.Query()`.

File: user_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func TestFollowUsersInvalidToken(t *testing.T) {
 	client := testClientString(http.StatusUnauthorized, json)
 	addDummyAuth(client)
 
-	err := client.Follow(ID("dummyID"))
+	err := client.FollowUser(ID("dummyID"))
```
Comment (zmb3) [inline user_test.go]: Could also add a test for `FollowArtist`

**Comment by zmb3** [comment]
Thanks again, looks great.

**Pull request merged**